src/shared/shared_ck/particle_dynamics/fluid_dynamics/acoustic_step_1st_half.h

```diff
@@ -148,6 +148,39 @@ class AcousticStep1stHalf<Contact<Wall, RiemannSolverType, KernelCorrectionType,
     FluidType &fluid_;
     RiemannSolverType riemann_solver_;
 };
+template <class RiemannSolverType, class KernelCorrectionType, typename... Parameters>
```

Comment:
The problem is a typical FSI problem, we do not need add extra method in fluid mechanics.